rama-tcp/src/server/listener.rs

```diff
@@ -93,8 +94,11 @@ where
     /// Binding with a port number of 0 will request that the OS assigns a port
     /// to this listener. The port allocated can be queried via the `local_addr`
     /// method.
-    pub async fn bind<A: ToSocketAddrs>(self, addr: A) -> io::Result<TcpListener<S>> {
-        let inner = TokioTcpListener::bind(addr).await?;
+    pub async fn bind<A: TryInto<SocketAddress, Error = io::Error>>(
```

Comment:
```suggestion
    pub async fn bind<A: TryInto<SocketAddress, Error: Into<BoxError>>>(
```

Probably you want something like this,
while at the same time adapting `io::Result<TcpListener<S>>` to `Result<TcpListener<S>, BoxError>`.

It will also mean that you need to chain `.map_err(Into::into) to `.bind(addr).await` as to make it type-match the err.

Reply:
Resolved in newest commit